### ISSUE TITLE
Paragraph tag + list (stability)

### ISSUE THREAD

**Issue by davidtorosyan** [tnorbye/kdoc-formatter#63]
When a paragraph tag is removed, unneeded newlines in a list aren't removed.

Before
```kt
/**
 * 1. Test
 *
 * <p>2. Test
 */
fun foo() = Unit
```

After:
```kt
/**
 * 1. Test
 *
 * 2. Test
 */
fun foo() = Unit
```

After again:
```kt
/**
 * 1. Test
 * 2. Test
 */
fun foo() = Unit
```

When using default options on v1.5.1.

**Comment by davidtorosyan** [comment]
Discussed over email with @tnorbye, the underlying issue is that kdocs don't want you to have to adjacent lists:
![image](https://user-images.githubusercontent.com/46736285/181064915-4104f6dd-1e79-4505-b151-23cf52c7a34a.png)

But actually, adding a `<p>` doesn't really fix that:
![image](https://user-images.githubusercontent.com/46736285/181065035-65d11ff7-fc78-44e7-a844-d007b8463ea6.png)

I think the right thing to do here would be to collapse the list (like is being done now), which would force the user to add a separator themselves if they want one:

![image](https://user-images.githubusercontent.com/46736285/181065203-5cd3e499-5142-41e5-a04b-70fbd65b21ca.png)

**Comment by tnorbye** [comment]
Implemented as David suggested above; it will now combine the lists directly. Fixed in 1.5.4

**Comment by davidtorosyan** [comment]
Reopening with another edge case, almost the same but a little different.

Before:
```kt
/**
 * Some title
 *
 * <p>1. Test
 * 2. Test
 */
fun foo() = Unit
```

After:
```kt
/**
 * Some title
 *
 * 1. Test
 * 2. Test
 */
fun foo() = Unit
```

After again:
```kt
/**
 * Some title
 * 1. Test
 * 2. Test
 */
fun foo() = Unit
```

Using v1.5.4 with default options.
